R/reporter-junit.R

```diff
@@ -0,0 +1,163 @@
+#' @include reporter.R
+NULL
+
+classnameOK <- function(text) {
+  gsub("[ \\.]", "_", text)
+}
+
+#' @importFrom xml2 xml_attr<- xml_add_child
+add_broken <- function (parent, type, message) {
+  child <- xml_add_child(parent, type)
+  xml_attr(child, 'type')    <- type
+  xml_attr(child, 'message') <- message
+}
+
+#' @importFrom xml2 xml_attr<-
+add_attrs <- function (node, ...) {
+  attrs <- list(...)
+  for (name in names(attrs)) {
+    xml_attr(node, name) <- as.character(attrs[[name]])
+  }
+}
+
+
+#' Test reporter: summary of errors in jUnit XML format.
+#'
+#' This reporter includes detailed results about each test and summaries,
+#' written to a file (or stdout) in jUnit XML format. This can be read by
+#' the Jenkins Continuous Integration System to report on a dashboard etc.
+#' Requires the XML package.
+#'
+#' This works for \code{\link{expect_that}} but not for the wrappers like
+#' \code{\link{expect_equal}} etc.
+#'
+#' To fit into the jUnit structure, context() becomes the \code{<testsuite>}
+#' name as well as the base of the \code{<testcase> classname}. The
+#' test_that() name becomes the rest of the \code{<testcase> classname}.
+#' The deparsed expect_that() call becomes the \code{<testcase>} name.
+#' On failure, the message goes into the \code{<failure>} node message
+#' argument (first line only) and into its text content (full message).
+#'
+#' Execution time and some other details are also recorded.
+#'
+#' References for the jUnit XML format:
+#' \url{http://llg.cubic.org/docs/junit/}
+#'
+#' Output drawn from the SummaryReporter is printed to the standard
```

Comment:
One reporter should do one thing, so I agree that you should drop this and simply output xml